tests/numpy/models/test_linear_model.py

```diff
@@ -0,0 +1,517 @@
+        )
+        w_ref = ridge_class.coef_[0][:, 0]
+        ridge_class_scaled = self.equisolve_solver_from_numpy_arrays(
+            scaling * X, scaling * y, property_w, scaling * sample_w
+        )
+        w_scaled = ridge_class_scaled.coef_[0][:, 0]
+
+        # Check that the two approaches yield the same result
+        assert_allclose(w_scaled, w_ref, atol=1e-11, rtol=1e-8)
+
+    def test_stability(self):
+        """Test numerical stability of the solver."""
+        pass
+
+    def test_scalar_alpha(self):
+        """Test ridge regression with scalar alpha."""
+        pass
+
+    def test_vector_alpha(self):
+        """Test ridge regression with a different alpha per property."""
+        pass
+
+    def test_sample_weights(self):
+        """Test ridge regression with a different value per target property."""
+        pass
+
+    @pytest.mark.parametrize(
+        "parameter_keys", [("values"), ("values", "positions"), ("positions")]
+    )
+    def test_parameter_keys(self, parameter_keys):
+        """Test regression with different parameter_keys.
+
+        Using a system with one atom per structure.
+        """
+        num_properties = 5
+        num_targets = 10
+        mean = 3.3
+
+        X_arr = np.random.normal(mean, 1, size=(4 * num_targets, num_properties))
+        w_exact = np.random.normal(mean, 3, size=(num_properties,))
+
+        # Create training data
+        X_values = X_arr[:num_targets]
+        X_block = matrix_to_block(X_values)
+
+        X_gradient_data = X_arr[num_targets:].reshape(num_targets, 3, num_properties)
+
+        position_gradient_samples = Labels(
+            ["sample", "structure", "atom"],
+            np.array([[s, 1, 1] for s in range(num_targets)]),
+        )
+
+        X_block.add_gradient(
+            parameter="positions",
+            data=X_gradient_data,
+            samples=position_gradient_samples,
+            components=[Labels(["direction"], np.arange(3).reshape(-1, 1))],
+        )
+
+        X = TensorMap(Labels.single(), [X_block])
+
+        # Create target data
+        y_arr = X_arr @ w_exact
+
+        y_values = y_arr[:num_targets].reshape(-1, 1)
+        y_block = matrix_to_block(y_values)
+
+        y_gradient_data = y_arr[num_targets:].reshape(num_targets, 3, 1)
+
+        y_block.add_gradient(
+            parameter="positions",
+            data=y_gradient_data,
+            samples=position_gradient_samples,
+            components=[Labels(["direction"], np.arange(3).reshape(-1, 1))],
+        )
+
+        y = TensorMap(Labels.single(), [y_block])
+
+        # Use no regularization.
+        alpha = tensor_to_tensormap(
+            np.zeros(num_properties).reshape(1, 1, -1), key_name="_"
+        )
+
+        clf = Ridge(parameter_keys=parameter_keys, alpha=alpha)
+        clf.fit(X=X, y=y)
+
+        assert_allclose(clf.coef_[0], w_exact.reshape(-1, 1), atol=1e-15, rtol=1e-8)
+
+        # Test prediction
+        X_pred = clf.predict(X)
+
```

Comment:
to check if two TensorMap are equal with also the same metadata you can use the operations.all_close(X_pred,y) function.